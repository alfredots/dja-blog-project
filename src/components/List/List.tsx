import { HTMLAttributes } from 'react';

import * as S from './styles';

type ListProps = HTMLAttributes<HTMLUListElement> & {
  children: React.ReactNode;
};

export const List = ({ children, ...rest }: ListProps) => {
  return (
    <>
      <S.Container {...rest}>{children}</S.Container>
    </>
  );
};
