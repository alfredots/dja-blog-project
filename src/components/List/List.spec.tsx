import { render, screen } from '@testing-library/react';

import { List } from './List';

describe('<List/>', () => {
  it('should render a list', () => {
    const children = 'children';

    const { container } = render(<List>{children}</List>);

    const childrenElement = screen.getByText(children);

    expect(childrenElement).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
