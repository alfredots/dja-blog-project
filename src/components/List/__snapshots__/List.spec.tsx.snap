// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List/> should render a list 1`] = `
<div>
  <ul
    class="flex justify-between gap-3"
  >
    children
  </ul>
</div>
`;
